docs_guidelines/tutorials/new_doc_project.rst

```diff
@@ -0,0 +1,610 @@
+        runs-on: ubuntu-latest
+        strategy:
+          matrix:
+            python-version: [3.8]
+        steps:
+
+Steps
+^^^^^
+
+Check-out the repository
+"""""""""""""""""""""""""
+
+The first step of this job consists of enabling the workflow to access the repository ``qiskit-x``. To do that you only need to call the latest version (version 3, ``@v3``) the pre-defined `checkout action <https://github.com/actions/checkout>`_ with the `uses <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses>`_ key.
+By setting the ``fetch-depth`` parameter to ``0`` via the `with <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idwith>`_ key, you can let the action reach for the full history of your repository.
+
+.. code-block:: yaml
+
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+
+Setup Python
+""""""""""""
+
+Once your repo has been reached, the workflow continues by installing a Python version, in this case, the 3.8, as defined in ``matrix.python-version``. Fortunately, the
+`setup-python action <https://github.com/actions/setup-python>`_ already does this for us!. To choose the Python version you only have to set the ``python-version`` parameter with the corresponding value.
+
+.. code-block:: yaml
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+
+Install the package
+""""""""""""""""""""
+
+Now that you have Python, let's install our package, Qiskit X and its more immediate dependencies and constraints from ``requirements-dev.txt`` and ``constraints.txt`` respectively. 
+You can use the `run <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun>`_ key and add the usual bash commands as a multiline block with the pipe indicator ``|``. These commands are: ``pip install -e .`` for the editable (``-e``) version of the package, ``pip install -U -r requirements-dev.txt`` for the latest version (``-U``) of the requirements (``-r``) and ``pip install -c constraints.txt`` for the constraints (``-c``).
+You can set the shell to ``bash`` via the
+`shell <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell>`_ keyword.
+
+
+.. code-block:: yaml
+
+    - name: Install X
+      run: |
+        pip install -e .
+        pip install -U -c constraints.txt -r requirements-dev.txt
+      shell: bash
+
+
+Install other dependencies
+""""""""""""""""""""""""""
+
+In case there are another dependencies that are not covered in the ``requirements-dev.txt``, you can add them in a similar step.
+
+.. code-block:: yaml
+
+    - name: Install Dependencies
+      run: |
+        pip install packages
+        sudo apt-get install -y other_packages
+      shell: bash
+
+Build and publish
```

Comment:
I think it would be good to link to the workflow file and `conf.py` in qiskit-meta and qiskit-nature (or any other applications modules) as examples. Perhaps you can add a reference section add this end and add these links.

Reply:
Added the ``conf.py``, ``deploy-docs.yml``/``docs.yml``, ``ignore_untagged_notes.sh`` and ``deploy_documentation.sh`` from the metapackage, nature and machine learning.